### PR TITLE
fix: NavigationStarted Event for sub-frames

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: tests/test_browsing_context.py [hunk 1]
```diff
@@ -963,8 +963,44 @@ async def test_browsingContext_fragmentNavigated_event(websocket, context_id,
 
 
 @pytest.mark.asyncio
-async def test_browsingContext_navigationStarted_event(websocket, context_id,
-                                                       html):
+@pytest.mark.parametrize("dataUrl", [True, False])
```
Comment (OrKoN) [inline tests/test_browsing_context.py:966]:
```suggestion
@pytest.mark.parametrize("url", ["https://example.com/", "data:text/html,"])
```

File: tests/test_browsing_context.py [hunk 2]
```diff
@@ -963,8 +963,44 @@ async def test_browsingContext_fragmentNavigated_event(websocket, context_id,
...
+async def test_browsingContext_navigationStartedEvent_viaScript(
+        websocket, context_id, html, dataUrl):
```
Comment (OrKoN) [inline tests/test_browsing_context.py:968]:
```suggestion
        websocket, context_id, html, url):
```

File: tests/test_browsing_context.py [hunk 3]
```diff
@@ -963,8 +963,44 @@ async def test_browsingContext_fragmentNavigated_event(websocket, context_id,
...
+
+    url = html() if dataUrl else "https://example.com/"
```
Comment (OrKoN) [inline tests/test_browsing_context.py:970]:
```suggestion
```

File: tests/test_browsing_context.py (hunk 2 above)
Comment (OrKoN) [inline tests/test_browsing_context.py:968]:
```suggestion
        websocket, context_id, dataUrl):
```

File: tests/test_browsing_context.py [hunk 4]
```diff
@@ -963,8 +963,43 @@ async def test_browsingContext_fragmentNavigated_event(websocket, context_id,
 
 
 @pytest.mark.asyncio
-async def test_browsingContext_navigationStarted_event(websocket, context_id,
-                                                       html):
+@pytest.mark.parametrize("url", ["https://example.com/", "data:text/html,"])
```
[the remark this answers is not in the document]
Comment (Lightning00Blade) [reply tests/test_browsing_context.py:966]: Prefer not to, as we that implies something else. `http://google.com` + `http://example.com` is valid values for  [`url_cross_origin`](https://github.com/GoogleChromeLabs/chromium-bidi/blob/0806ee900d1cdaffb9bf6de26078f94a00f2f6dc/tests/conftest.py#L111)

**Review by OrKoN — APPROVED** [review]

**Comment by thiagowfx** [comment]
Is this part of https://github.com/GoogleChromeLabs/chromium-bidi/issues/1018?

If yes, then please add `Bug: #1018` to the PR desc.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**